File: app/src/main/java/io/mountx/wutai/feature/objects/ObjectsAdapter.java
package io.mountx.wutai.feature.objects;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

/**
 * @author hhs
 * Created on 2018/10/22
 */
class ObjectsAdapter extends FragmentStatePagerAdapter {

    private final String[] titles = {"expression", "singleton", "companion", "jvmStatic"};

    public ObjectsAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        switch (i) {
            case 0:
                return ObjectExpressionFragment.Companion.newInstance();
            case 1:
                return SingletonFragment.Companion.newInstance();
            case 2:
                return CompanionObjectFragment.Companion.newInstance();
            default:
                return JvmStaticFragment.newInstance();
        }
    }

    @Override
    public int getCount() {
        return titles.length;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }

}

File: app/src/main/java/io/mountx/wutai/feature/objects/ObjectsAdapter.kt
package io.mountx.wutai.feature.objects

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

/**
 * @author Ha Sang
 * Created on 2018/10/22
 */
class ObjectsAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val titles = arrayOf("expression", "singleton", "companion", "jvmStatic")

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> ObjectExpressionFragment.newInstance()
            1 -> SingletonFragment.newInstance()
            2 -> CompanionObjectFragment.newInstance()
            else -> JvmStaticFragment.newInstance()
        }
    }

    override fun getCount(): Int {
        return titles.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titles[position]
    }
}
